javascript.md

```diff
@@ -1156,9 +1334,16 @@ console.log(props); // outputs {y: "hello", "z": hello}
 Rust:
 
 ```rust
-// TODO
+struct Whatever { x: String, y: String, z: String }
+let whatever = Whatever { x: "hi".into(), y: "hello".into(), z: "hello".into() };
+let Whatever { x, .. } = whatever;
+println!("{}", x); // outputs "hi"
 ```
 
+Note: You can't destructure using `..foo` and get an object with `y` and `z` fields,
+because Rust doesn't have anonymous types like that (see the Object Literals section)
+You _can_ use `..` to discard the other fields when destructuring, though.
+
 ## Array comprehension
```

Comment:
JS doesn't do array comprehension either (only non-standard Mozilla experiment), so we should ditch this.